File: src/topic/scripts/person_talker.py

import rospy
from topic.msg import Person


if __name__ == "__main__":

    rospy.init_node("talker_person_p")
    pub = rospy.Publisher("chatter_person",Person,queue_size=10)
    p = Person()
    p.name = "葫芦娃"
    p.age = 18
    p.height = 0.75

    rate = rospy.Rate(1)
    while not rospy.is_shutdown():
        pub.publish(p)  
        rate.sleep()  
        rospy.loginfo("姓名:%s, 年龄:%d, 身高:%.2f",p.name, p.age, p.height)

File: src/topic/src/person_talker.cpp
#include "ros/ros.h"
#include "topic/Person.h"

int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");

    ros::init(argc,argv,"talker_person");

    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<topic::Person>("chatter_person",1000);

    topic::Person p;
    p.name = "sunwukong";
    p.age = 2000;
    p.height = 1.45;

    ros::Rate r(1);
    while (ros::ok())
    {
        pub.publish(p);
        p.age += 1;
        ROS_INFO("我叫:%s,今年%d岁,高%.2f米", p.name.c_str(), p.age, p.height);

        r.sleep();
        ros::spinOnce();
    }

    return 0;
}